rust/src/dns/dns.rs

```diff
@@ -122,18 +124,94 @@ pub const DNS_RCODE_BADTRUNC: u16 = 22;
 /// gets logged.
 const MAX_TRANSACTIONS: usize = 32;
 
+static mut ALPROTO_DNS: AppProto = ALPROTO_UNKNOWN;
+
 #[repr(u32)]
 pub enum DNSEvent {
-    UnsolicitedResponse = 0,
-    MalformedData,
-    NotRequest,
-    NotResponse,
-    ZFlagSet,
-    Flooded,
-    StateMemCapReached,
+    MalformedData = 0,
+    NotRequest = 1,
+    NotResponse = 2,
+    ZFlagSet = 3,
+}
+
+impl DNSEvent {
+    pub fn to_cstring(&self) -> &str {
+        match *self {
+            DNSEvent::MalformedData => "MALFORMED_DATA\0",
```

Comment:
off topic: can we have a macro to automatically add \0 to strings like this? `my_macro!("MALFORMED_DATA")` which would lead to `"MALFORMED_DATA\0"`?

Reply:
Possible, but I'd like to put that thought on hold.  I have another branch where we derive most of this event code, so the writer of the code doesn't have to add the \0 as well.

Reply:
Ack.